src/testdrive/src/main.rs

```diff
@@ -165,14 +172,50 @@ async fn run(args: Args) -> Result<(), Error> {
     if args.files.is_empty() {
         testdrive::run_stdin(&config).await
     } else {
+        let mut errors = Vec::new();
+
         for file in args.files {
-            if file == "-" {
-                testdrive::run_stdin(&config).await?
-            } else {
-                testdrive::run_file(&config, &file).await?
+            if let Err(error) = match file.as_str() {
+                "-" => testdrive::run_stdin(&config).await,
+                _ => testdrive::run_file(&config, &file).await,
+            } {
+                let _ = error.print_stderr(args.ci_output);
+                errors.push(error);
+
+                if errors.len() >= args.max_errors {
+                    break;
+                }
+            }
+        }
+
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            let mut error_files = Vec::new();
+            for error in &errors {
+                if let Error::Input {
+                    details: Some(details),
+                    ..
+                } = error
+                {
+                    if !error_files.contains(&details.filename()) {
+                        error_files.push(details.filename());
+                    }
+                }
             }
+
+            let mut hints = vec![format!("error count: {}", errors.len())];
+
+            if !error_files.is_empty() {
+                hints.push(format!("files involved: {}", error_files.join(" ")));
+            }
+
+            Err(Error::General {
+                ctx: "Errors were encountered during execution".into(),
+                causes: Vec::new(),
+                hints,
+            })
```

Comment:
So the structure of this whole block is a little funky! The same multi-error handling should apply to the `if` block too (i.e., when no files are specified so the file is implicitly stdin). Then matching for `Error::Input` to pull out the filename is overly restrictive; since you know what file errored, you can just put file into `error_files` directly.